gradle-plugin-testing-junit/src/main/java/com/palantir/gradle/testing/execution/GradleInvocation.java

```diff
@@ -32,10 +40,60 @@ public GradleInvocation withEnvironment(Map<String, String> environment) {
     }
 
     public InvocationResult buildsSuccessfully() {
-        return new InvocationResult(gradleRunner.build());
+        InvocationResult result = new InvocationResult(gradleRunner.build());
+
+        if (configurationCacheEnabled) {
+            assertConfigCacheStored(result.output());
+
+            // Run a second time to verify the cache is reused, we don't need to run the tasks --dry-run is enough to
+            // check if the configuration cache was used.
+            GradleRunner configurationCacheRunner = GradleRunner.create()
+                    .withProjectDir(gradleRunner.getProjectDir())
+                    .withDebug(gradleRunner.isDebug())
+                    .forwardOutput()
+                    .withGradleVersion(gradleVersion.version())
+                    .withPluginClasspath(gradleRunner.getPluginClasspath())
+                    .withEnvironment(gradleRunner.getEnvironment())
+                    .withArguments(ImmutableList.<String>builder()
+                            .addAll(gradleRunner.getArguments())
+                            .add("--dry-run")
```

Comment:
for config-cache to be reused, we need to run the same tasks. In order to avoid running the same tasks twice, we can pass a `--dry-run`. 
Tested that config cache is triggered by a dry-run: 

```
Reusing configuration cache.
:compileJava SKIPPED
:help SKIPPED

BUILD SUCCESSFUL in 62ms
Configuration cache entry reused.
```